src/main/java/com/akademia/educators_back/service/impl/ProblemServiceImpl.java

```diff
@@ -6,32 +6,46 @@
 import com.akademia.educators_back.exception.ProblemDoesNotExistException;
 import com.akademia.educators_back.mapper.ProblemMapper;
 import com.akademia.educators_back.repository.ProblemRepository;
+import com.akademia.educators_back.valicators.ProblemValidator;
+import lombok.AllArgsConstructor;
 import org.springframework.stereotype.Service;
 
 import java.util.ArrayList;
 import java.util.List;
 
 @Service
+@AllArgsConstructor
 public class ProblemServiceImpl implements Problem {
-
     private ProblemRepository problemRepository;
     private ProblemMapper problemMapper;
-    private ProblemEntity problemEntity;
+    private ProblemValidator problemValidator;
+
     @Override
     public void addProblemToDB(ProblemTo problemTo) {
-        problemEntity = problemMapper.toProblemEntity(problemTo);
+        problemValidator.problemCategoryExistingChecking(problemTo);
+        problemValidator.problemTitleLengthChecking(problemTo);
+        problemValidator.problemQuestionExistingChecking(problemTo);
+        problemValidator.problemQuestionLengthChecking(problemTo);
+        ProblemEntity problemEntity = problemMapper.toProblemEntity(problemTo);
         problemRepository.save(problemEntity);
     }
 
     @Override
     public void deleteProblemFromDB(ProblemTo problemTo) {
-        problemEntity = problemMapper.toProblemEntity(problemTo);
+        problemValidator.problemCategoryExistingChecking(problemTo);
+        problemValidator.problemTitleLengthChecking(problemTo);
+        problemValidator.problemQuestionExistingChecking(problemTo);
+        ProblemEntity problemEntity = problemMapper.toProblemEntity(problemTo);
         problemRepository.delete(problemEntity);
     }
 
     @Override
     public void updateProblem(ProblemTo problemTo) {
-        problemEntity = problemMapper.toProblemEntity(problemTo);
+        problemValidator.problemCategoryExistingChecking(problemTo);
```

Comment:
to wszystko związane z walidacją wyciągnąć do osobnej metody prywatnej na przykład

Reply:
to samo w całym serwisie ;)